packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/library/dto/PublishLibrariesInput.java

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.library.dto;
+
+import java.util.UUID;
+
+import org.eclipse.sirius.components.core.api.IInput;
+
+import jakarta.validation.constraints.NotNull;
+
+/**
+ * Input used to publish libraries.
+ *
+ * @author gdaniel
+ */
+public record PublishLibrariesInput(@NotNull UUID id, @NotNull String projectId, @NotNull String publicationKind, @NotNull String version, @NotNull String description) implements IInput {
```

Comment:
We should probably replace `projectId` with `editingContextId` to let users publish libraries from any editing context of their project

Reply:
I don't see us publish libraries from anything else anytime soon and having the projectId directly is quite helpful so I like it but this decision could be challenged later.